app/src/main/java/com/example/mycalculator/mvp/model/CalculatorModel.java

```diff
@@ -28,52 +32,69 @@ public void saveNumber(String number) {
+            if (firstOperand.isEmpty()){
+                result = EMPTY_STRING;
+            } else {
+                result = String.valueOf(Double.parseDouble(firstOperand));
+            }
+            return false;
+        }else if (secondOperand.isEmpty()){
+            result = ERROR_MESSAGE_INVALID_FORMAT;
+            return false;
+        }
+        return true;
+    }
+
     @Override
     public String getFullResult() {
 
-        switch (operator) {
+         if (isValidOperation()) {
 
-            case OPERATOR_SUM:
-                if (result == EMPTY_STRING) {
-                    result = String.valueOf(Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand));
-                }
+            switch (operator) {
 
-                break;
+                case OPERATOR_SUM:
+                        result = String.valueOf(Double.parseDouble(firstOperand) + Double.parseDouble(secondOperand));
+                        break;
 
-            case OPERATOR_MINUS:
-                result = String.valueOf(Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand));
-                break;
+                case OPERATOR_MINUS:
+                    result = String.valueOf(Double.parseDouble(firstOperand) - Double.parseDouble(secondOperand));
+                    break;
 
-            case OPERATOR_DIVIDE:
-                result = String.valueOf(Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand));
-                break;
+                case OPERATOR_DIVIDE:
+                    if ((Double.parseDouble(secondOperand)) == Double.parseDouble(TEXT_ZERO_NUMBER)) {
+                        result = ERROR_MESSAGE_DIVISION;
+                    } else {
+                        result = String.valueOf(Double.parseDouble(firstOperand) / Double.parseDouble(secondOperand));
+                    }
+                    break;
 
-            case OPERATOR_MULTIPLY:
-                result = String.valueOf(Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand));
-                break;
+                case OPERATOR_MULTIPLY:
+                    result = String.valueOf(Double.parseDouble(firstOperand) * Double.parseDouble(secondOperand));
+                    break;
 
-            default:
-                result = ERROR_MESSAGE;
+                default:
+                    result = ERROR_MESSAGE;
+            }
+        }
+            return result;
         }
 
-        return result;
-    }
-
-    @Override
-    public String eraseResult() {
-        result = EMPTY_STRING;
-        firstOperand = EMPTY_STRING;
-        secondOperand = EMPTY_STRING;
-        operator = EMPTY_STRING;
-        return EMPTY_STRING;
-    }
+        @Override
+        public String eraseResult () {
+            result = EMPTY_STRING;
+            firstOperand = EMPTY_STRING;
+            secondOperand = EMPTY_STRING;
+            operator = EMPTY_STRING;
+            return EMPTY_STRING;
```

Comment:
why in this method you return a String? is really necessary?

Reply:
i used a string since all values ​​are of that type, and then parsed to double. What I am looking for is that when pressing the delete key, all the values ​​return to the initial state.